euphoria-core/src/main/java/cz/seznam/euphoria/core/client/io/Context.java

```diff
@@ -16,27 +16,8 @@
 package cz.seznam.euphoria.core.client.io;
 
 /**
- * A collector of elements. Used in functors.
- *
- * @param <T> the type of elements collected through this context
+ * Used in user defined functions to access environment methods.
  */
-public interface Context<T> {
-
-  /**
-   * Collects the given element to the output of this context.
-   *
-   * @param elem the element to collect
-   */
-  void collect(T elem);
-
-  /**
-   * Retrieves the window - if any - underlying the current
-   * execution of this context.
-   *
-   * @return {@code null} if this context is not executed within a
-   *          windowing strategy, otherwise the current window of
-   *          this context
-   */
-  Object getWindow();
+public interface Context extends Environment {
```

Comment:
Good question. It is the same. Basically the `Collector` should be directly implementing `Context`. It means collector just extends the context with a possibility to emit values.
The reason there is no direct relation between `Collector` and `Context` are lambda expressions that could become ambiguous and couldn't be used without casting.
```java
interface UnaryFunctor<IN, OUT> {
   apply(IN elem, Collector<OUT> collector);
}

UnaryFunctionEnv<IN, OUT> {
    OUT apply(IN what, Context context);
}
````